src/python/pants/build_graph/target_filter_subsystem.py

```diff
@@ -0,0 +1,41 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+from pants.subsystem.subsystem import Subsystem
+
+
+logger = logging.getLogger(__name__)
+
+
+class TargetFilter(Subsystem):
+  """Filter out targets matching given options.
+
+  :API: public
+  """
+
+  options_scope = 'target-filter'
+
+  @classmethod
+  def register_options(cls, register):
+    super(TargetFilter, cls).register_options(register)
+
+    register('--exclude-tags', type=list,
+             default=[],
```

Comment:
I think this will be True for all options in the new engine, but I believe this option should be `fingerprint=True`. If the tags change, the subsystem instance fundamentally changes, producing new outputs for the same `targets` inputs.